### DISCUSSION TITLE
Demo: don't redirect share links to the logged-in view

### DISCUSSION THREAD

**Discussion by strugee** [photoprism/photoprism#3550] [General]
**Describe what problem this solves and why this would be valuable to many users**

demo.photoprism.app seems to have code that automatically logs you in when you load it. However, this logic is applied to share links as well. For example, `https://demo.photoprism.app/s/1in091ddw0/share-bug-test` (I assume this will expire shortly but I'm including it anyway to make clear which URL route I'm talking about). This link was shared from a newly-created album.

This has two important consequences:

1. It's impossible to see a demo of what album/etc. shares look like as a prospective user
2. You can't reproduce share bugs in the demo environment - particularly useful if you want to confirm the bug you've found hasn't been fixed in a newer PhotoPrism version

**Describe the solution you'd like**

demo.photoprism.app shouldn't automatically log you in if the URL is a share link. Unclear to me if it should always display the public view or only if the user hasn't automatically been logged in already - both would make the share link appear as it would in a non-demo environment in a private browsing window, but the latter would match non-demo behavior more closely in regular browsing windows too.

**Describe alternatives you've considered**

See above.

**Additional context**

N/A

**Comment by lastzero** [comment]
It's expected because our demo is running in public mode. We also have a non public demo available though.

**Comment by strugee** [comment]
Hrm, where is it? I couldn't find any references to it offhand - maybe I'm missing somewhere obvious though.

**Comment by lastzero** [comment]
@strugee I've emailed you the login details for our private demo :)
